Ical.Net/DataTypes/Attachment.cs

```diff
@@ -93,15 +94,15 @@ public override void CopyFrom(ICopyable obj)
         FormatType = att.FormatType;
     }
 
-    protected bool Equals(Attachment other)
+    protected bool Equals(Attachment? other)
     {
-        var firstPart = Equals(Uri, other.Uri) && ValueEncoding.Equals(other.ValueEncoding);
+        var firstPart = Equals(Uri, other?.Uri) && ValueEncoding.Equals(other?.ValueEncoding);
         return Data == null
             ? firstPart
-            : firstPart && Data.SequenceEqual(other.Data);
+            : firstPart && other?.Data != null && Data.SequenceEqual(other.Data);
```

Comment:
This comparison is not correct. If `other?.Data == null`, then `this.Data` is ignored, even if it contains data.